torch/csrc/autograd/functions/batch_normalization.h

```diff
@@ -34,12 +34,15 @@ struct BatchNormBackward : public Function, public BatchNormParams {
       SavedVariable weight,
       SavedVariable bias)
     : Function(std::move(flags))
-    , BatchNormParams(std::move(params))
-    , save_mean(std::move(save_mean))
-    , save_std(std::move(save_std))
-    , input(std::move(input))
-    , weight(std::move(weight))
-    , bias(std::move(bias)) {}
+    , BatchNormParams(std::move(params)) {
+      if (is_executable) {
+        this->save_mean = std::move(save_mean);
```

Comment:
Because you don't want to save anything if the function is not going to be executed. Otherwise it's going to be wasting a lot of memory. It was the cause of that recent memory regression.